src/crimson/osd/pg_recovery.cc

```diff
@@ -150,6 +150,9 @@ size_t PGRecovery::start_primary_recovery_ops(
     if (!pg->get_recovery_backend()->is_recovering(soid)) {
       if (pg->get_recovery_backend()->is_recovering(head)) {
 	++skipped;
+	auto& recovery_waiter = pg->get_recovery_backend()->get_recovering(soid);
+	out->push_back(recovery_waiter.wait_for_recovered_blocking());
+	started++;
```

Comment:
I don't really understand what's going on here, why do we both increment skipped and started?  Why do we only care if we're not recovering soid, but are recovering head?